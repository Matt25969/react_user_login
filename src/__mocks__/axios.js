
module.exports = {
  get: (url) => {
      if (url === "http://localhost:5000/user/name/valid/valid"){
    return Promise.resolve({data:{Status : "Logged In"}});
  }
  else
  {
    return Promise.resolve({data:{Status : "Not Logged In"}});
  }
  },


  post: (url, body) => {

  if (body.username==='invalid'){
  
  return Promise.resolve({data:{username:"Username field is required"}});
  }
  if (body.email==='invalid'){
  
    return Promise.resolve({data:{email:"Email field is required"}});
    }
}

  
};