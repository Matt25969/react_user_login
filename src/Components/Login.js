import React, { Component } from 'react';
import _ from 'lodash'
import axios from 'axios'

import '../App.css';

export default class Login extends Component {

    constructor(props) {
        super(props);
        this.state = {
            log:"Not Logged In"
        }
    }

    handleSubmit = (e) => {

        e.preventDefault();

       axios.get("http://localhost:5000/user/name/"+e.target[0].value+"/"+e.target[1].value).then(res => {
          
                this.setState({
                    log:res.data.Status
                })
        })

    }

render() {
  return (
    <div>
        <form id="LoginForm" onSubmit={this.handleSubmit}>
<input placeholder="username"></input>
<br></br>
<input placeholder="password"></input>
<br></br>
<input type="submit"></input>
</form>
<p id="result">{this.state.log}</p>
    </div>
  );
}
    }
