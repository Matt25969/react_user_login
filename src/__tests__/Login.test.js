import React from 'react';
import axios from 'axios';

import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16';
import {  mount, configure } from 'enzyme';

import Login from '../Components/Login.js'

jest.mock('axios');

configure({ adapter: new Adapter() });

it('renders Login comp', () => {
  const tree = renderer.create(<Login />).toTree;
  
  expect(tree).toMatchSnapshot();
})


it('Button submits form with valid data', () => {

  const getSpy = jest.spyOn(axios, 'get');

  const context = {};

  const wrapper = mount(<Login />, { context });

    const e = {
        target: [{
            value: "valid"
        },
        {
            value: "valid"
        }]
    };

    wrapper.find('[id="LoginForm"]').last().simulate("submit", e);

    const getPromise = getSpy.mock.results.pop().value;

    return getPromise.then(() => {
      
      const currentState = wrapper.state();

      wrapper.setContext({});

      expect(wrapper.contains(<p id="result">Logged In</p>)).toBe(true);
      
      expect(currentState.log.includes(("Logged In"))).toBe(true);
      
    })

});

it('Button submits form with invalid data', () => {

  const getSpy = jest.spyOn(axios, 'get');

  const context = {};

  const wrapper = mount(<Login />, { context });

    const e = {
        target: [{
            value: "invalid"
        },
        {
            value: "invalid"
        }]
    };

    wrapper.find('[id="LoginForm"]').last().simulate("submit", e);

    const getPromise = getSpy.mock.results.pop().value;

    return getPromise.then(() => {
      
      const currentState = wrapper.state();

      wrapper.setContext({});

      expect(wrapper.contains(<p id="result">Not Logged In</p>)).toBe(true);
      
      expect(currentState.log.includes(("Not Logged In"))).toBe(true);

    })

});