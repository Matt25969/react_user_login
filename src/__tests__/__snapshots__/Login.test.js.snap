// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button submits form 1`] = `ReactWrapper {}`;

exports[`renders Login comp 1`] = `[Function]`;
